sound/soc/intel/boards/sof_maxim_common.c

```diff
@@ -117,13 +148,26 @@ static int max_98373_hw_params(struct snd_pcm_substream *substream,
 				return -EINVAL;
 			}
 
+			/* get the tx mask from ACPI device properties */
+			tx_mask = max_98373_get_tx_mask(codec_dai->dev);
+			if (!tx_mask)
+				return -EINVAL;
+
+			if (tx_mask & tx_mask_used) {
+				dev_err(codec_dai->dev, "invalid tx mask 0x%x, used 0x%x\n",
+					tx_mask, tx_mask_used);
+				return -EINVAL;
+			}
+
+			tx_mask_used |= tx_mask;
+
```

Comment:
It's an interesting suggesting, but do we know for sure that ALL platforms based on max_98373 expose those properties? It might be safer if we don't to keep the previous hard-coded 3 as a default?

Or put differently, do we know of any platforms where the TX mask 3 is a problem? It would be interesting to expand on the problem statement and why moving to ACPI properties is needed now.

Reply:
Codec driver needs these two properties to setup which channel for i-sense and which for v-sense. And the purpose of tx_mask here is to setup which channels to output data and others to keep Hi-Z.

Considering that DSM is running in SOF DSP side and it requires iv-sense data from codec, those two properties should be set properly in ACPI table to make the speaker path working.

There is high consistency between Chromebook designs because they all share same coreboot code base. So I think there would be no problem now without this PR, even in the future. I think the biggest benefit is to improve some readability,

Reply:
Right, but what you forgot to mention is that if the properties are not defined the fallback is the same value as the existing hard-coded mask so there's no regression, only the capability to use a different mask if Coreboot was updated.